pypsa/optimization/constraints.py

```diff
@@ -105,16 +154,391 @@ def define_operational_constraints_for_extendables(
         lhs_upper += ((1, loss),)
 
     n.model.add_constraints(
-        lhs_lower, ">=", 0, name=f"{c}-ext-{attr}-lower", mask=active
+        lhs_lower, ">=", 0, name=f"{c}-ext-comm-non-mod-{attr}-lower", mask=active
+    )
+    n.model.add_constraints(
+        lhs_upper, "<=", 0, name=f"{c}-ext-comm-non-mod-{attr}-upper", mask=active
+    )
+
+
+def define_committability_variables_constraints_for_non_modular(n, sns, c, attr):
```

Comment:
I'd recommend to have a single function that takes care of setting the status variable for all components; there is quite a strong duplication among them.
The more general function is basically equivalent to the modular one, where the non-modular items get a 1.

Reply:
In previous comments I explained the need for 3 functions in the definition of constraints. But to summarise, from my point of view we have 3 different situations:
a) status variable <= n_mod, where n_mod is a variable and therefore belongs to lhs
b) status variable <= P_nom/P_nom_mod, where element is not extensible and P_nom/P_nom_mod belongs to rhs
c) status variable <= 1, for elements for which p_nom_mod is not declared, regardless of whether the element is extendable or not (since event for extendable elements, where p_nom_mod is not declared, but we have to use 1 as upper limit). 

Cases b) and c) can be combined into one, but looking at other functions it looks to me preferable to separate the cases. Of course, another way to reduce the functions would be to select the IDX to which the function is to be applied outside the function itself, but in actually used functions the selection of the cases is done in the function, so I kept this logic. What do you think?